internal/pkg/template/template_functions.go

```diff
@@ -25,6 +26,41 @@ func ReplaceDashesFunc(logicalID string) string {
 	return strings.ReplaceAll(logicalID, "-", dashReplacement)
 }
 
+// IsArnFunc takes a string value and determines if it's an ARN or not
+func IsARNFunc(value string) bool {
+	return arn.IsARN(value)
+}
+
+// IsSSMParamFunc takes a string value and determines if it's a SSM parameter
+func IsSSMParamFunc(value string) bool {
+	return strings.Contains(value, ":ssm:") || !strings.Contains(value, ":secretsmanager:")
+}
+
+// IsSecretsManagerSecretsFunc takes a string value and determines if it's a secretes manager secret
+func IsSecretsManagerSecretsFunc(value string) bool {
+	return strings.Contains(value, ":secretsmanager:")
+}
+
+// IsSSMParamFlagFunc takes secret values and determines if it consist a SSM parameter
+func IsSSMParamFlagFunc(secrets map[string]string) bool {
+	for _, value := range secrets {
+		if !arn.IsARN(value) || strings.Contains(value, ":ssm:") {
```

Comment:
We can perhaps reuse the other functions, e.g.
```
if !IsARNFunc(value) || IsSSMPramFunc(value)
```

Reply:
I have removed all the public functions as we now have SSMParamSecrets and secretsManagerSecrets which helps us find out if those particular secrets are present or not.